File: KiwiJuiceEasy.php
<?php
class KiwiJuiceEasy
{
    public function __construct()
    {
    }

    public static function instance()
    {
        return new self();
    }

    public function thePouring_(Array $capacities, Array $bottles, Array $fromId, Array $toId)
    {
        for ($i = 0; $i < count($fromId); $i++) {
            while ($bottles[$i] > 0) {
                if ($capacities[$i] <= $bottles[$i]) break;
                $bottles[$fromId[$i]]--;
                $bottles[$toId[$i]]++;
            }
        }
        return $bottles;
    }

    public function thePouring(Array $capacities, Array $bottles, Array $fromId, Array $toId)
    {
        for ($i = 0; $i < count($fromId); $i++) {
            $diff = min($capacities[$toId[$i]] - $bottles[$toId[$i]], $bottles[$fromId[$i]]);
            if ($diff <= 0) continue;
            $bottles[$fromId[$i]] = $bottles[$fromId[$i]] - $diff;
            $bottles[$toId[$i]] = $bottles[$toId[$i]] + $diff;
        }
        return $bottles;
    }
}

File: KiwiJuiceEasy.rb
class KiwiJuiceEasy
  def thePouring(capacities, bottles, from_id, to_id)
    from_id.each_with_index do |v, k|
      diff = [(capacities[to_id[k]] - bottles[to_id[k]]), bottles[from_id[k]]].min
      next if diff == 0
      bottles[from_id[k]] -= diff
      bottles[to_id[k]] += diff
    end
    bottles
  end
end
